### PR TITLE
sqlite: add NMake build system support

### PR DESCRIPTION
**Pull request by johnwparent**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sqlite/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/sqlite/package.py]: Needs `freebsd`.

Instead of these loops, let's get #41453 merged first.

Comment (johnwparent) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Oh geez, I apparently missed us adding `freebsd` somehow... thanks for pointing that out.

A. huge fan of #41453, thanks for adding that, these loops have been annoying me since day 1.
B. I'd prefer to let this land naturally and then go back and refactor after #41453 lands. #41453 hasn't received attention since the PR was made, this package is in immediate demand, and I don't want it to bit-rot here over the holidays. I'll have to refactor at least half a dozen packages after #41453 lands anyway, so this will just be one more.

Comment (haampie) [reply var/spack/repos/builtin/packages/sqlite/package.py]: That was _very_ recent ;p two week ago?

Comment (johnwparent) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Ok, I guess I didn't really look at how long ago the PR was made, just that there were no recent updates. You're right, two weeks isn't that long, but given that your change is non trivial, and this is somewhat closer to that, I'd still prefer to not hold this up.

**Review by scheibelp — DISMISSED** [review]

File: var/spack/repos/builtin/packages/sqlite/package.py
Comment (scheibelp) [reply var/spack/repos/builtin/packages/sqlite/package.py]: I'm also a fan of merging this before https://github.com/spack/spack/pull/41453, but this should add `freebsd` to the list before it is merged.

**Review by scheibelp — APPROVED** [review]
(I have now updated the everywhere-but-windows check to include freebsd)

**Comment by haampie** [comment]
This failed in all gitlab pipelines https://gitlab.spack.io/spack/spack/-/pipelines/585436 but status didn't come through?

**Comment by haampie** [comment]
Please test on Linux whenever you do anything Windows. Sure it would've been caught if Gitlab CI wasn't broken, but I feel like I have to repeat this every month.

**Pull request merged**
